File: Proyectos Android/NavigationPractica/app/src/main/java/com/example/navigationpractica/MainActivity.java
package com.example.navigationpractica;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.navigation.NavController;
import androidx.navigation.fragment.NavHostFragment;
import androidx.navigation.ui.AppBarConfiguration;
import androidx.navigation.ui.NavigationUI;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;

import com.example.navigationpractica.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {
    private ActivityMainBinding binding;
    private NavController navController;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        setSupportActionBar(binding.toolbar);

        //Para que muestre el menu del AppBar hay que darle los layouts desde los que será visible el menú,
        //en el resto se dará la opcion de volver al primer layout (Lo que me ha costado  darme cuenta de esto es bárbaro)
        AppBarConfiguration appBarConfiguration = new AppBarConfiguration.Builder(R.id.tabFragment, R.id.drawer2Fragment)
                .setOpenableLayout(binding.drawerLayout)
                .build();
        navController = ((NavHostFragment) getSupportFragmentManager().findFragmentById(R.id.fragmentContainerView)).getNavController();
        //Estableciendo la navegación en el Bottom Navigation View
        NavigationUI.setupWithNavController(binding.bottomNavView, navController);
        //Estableciendo la navegación en el Navigation View
        NavigationUI.setupWithNavController(binding.navView, navController);
        //Estableciendo la navegación en el toolbar
        NavigationUI.setupWithNavController(binding.toolbar, navController, appBarConfiguration);

    }

    //Inflando el menú de opciones
    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.options_menu, menu);
        return super.onCreateOptionsMenu(menu);
    }

    //Configurando la selección de elementos del Options Menu
    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        return NavigationUI.onNavDestinationSelected(item, navController) || super.onOptionsItemSelected(item);
    }
}

File: Proyectos Android/NavigationPracticaKT/app/src/main/java/com/example/navigationpracticakt/MainActivity.kt
package com.example.navigationpracticakt

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import androidx.navigation.NavController
import androidx.navigation.fragment.NavHostFragment
import androidx.navigation.ui.AppBarConfiguration
import androidx.navigation.ui.NavigationUI
import com.example.navigationpracticakt.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var navController: NavController

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        setSupportActionBar(binding.toolbar)

        //For the AppBar's Menu to appear we have to give it the layouts from which it will be visible
        //on the rest it will give the option to return to the main layout
        val appBarConfig = AppBarConfiguration.Builder(R.id.tabFragment, R.id.drawer1Fragment, R.id.drawer2Fragment)
            .setOpenableLayout(binding.drawerLayout).build()
        navController = (supportFragmentManager.findFragmentById(R.id.fragmentContainerView) as NavHostFragment).navController
        //Establishing the navigation to Bottom Navigation View
        NavigationUI.setupWithNavController(binding.bottomNavView, navController)
        //Establishing the navigation to Navigation View
        NavigationUI.setupWithNavController(binding.navView, navController)
        //Establishing the navigation to Toolbar
        NavigationUI.setupWithNavController(binding.toolbar, navController, appBarConfig)
    }

    //Inflating option's menu
    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.options_menu, menu)
        return super.onCreateOptionsMenu(menu)
    }

    //Configuring the element selection in Options Menu
    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return NavigationUI.onNavDestinationSelected(item, navController) ||
                super.onOptionsItemSelected(item)
    }
}